Added migration docs (#1442)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/migrate-to-okta/migrate-users-password-hooks/index.md

---
title: Import Users with Inline Password Hooks
---

## Migration Program Plan

> A migration program utilizes Okta's Inline Password Hook feature to seamlessly migrate users as they authenticate. The broad strokes look like:

Reviewer: ```suggestion
A migration program utilizes Okta's Inline Password Hook feature to seamlessly migrate users as they authenticate. The broad strokes look like this:
```

> 1. Create all the users from the legacy system in Okta with a provider set to: `IMPORT`. **NOTE:** This can be done in bulk and does NOT require individual user credentials.

Reviewer: ```suggestion
1. Create all the Users from the legacy system in Okta with a provider set to: `IMPORT`. Note that this can be done in bulk and does not require individual user credentials.
```

> 2. Optional: Create groups and applications in Okta. Assign users to groups and assign groups and users to applications

Reviewer: ```suggestion
2. Optional: Create Groups and Applications in Okta. Assign Users to Groups and assign Groups and Users to Applications
```

3. Create an application to service requests from Okta that can validate credentials against the legacy user system.
4. Register an Inline Password Hook with Okta that connects to the application created in step 3.

> Typically, you run a migration program like this for a set period of time, say 60 days. At the end of that time, you delete the inline password hook. For any users that have not completed the migration on their own, you can issue a password reset using the Okta API. Those users would receive an email to change their password and would then be able to login using Okta.

Reviewer: ```suggestion
Typically, you run a migration program like this for a set period of time, say 60 days. At the end of that time, you delete the Inline Password Hook. For any users that have not completed the migration on their own, you can issue a password reset using the Okta API. Those users would receive an email to change their password and would then be able to sign in using Okta.
```
